docs/src/modules/concepts/pages/architecture-model.adoc

```diff
@@ -1,24 +1,44 @@
+Akka encourages a project structure that separates your system’s Application Programming Interfaces (APIs), Akka component logic, and business logic into different directories.
 
-== Layered structure
+This structure supports a clear separation of concerns. It helps enable iterative development, testing in isolation, predictable packaging, and the ability to externalize configuration and static assets.
 
-Akka services follow a layered model where each layer has a distinct role. The layers are arranged concentrically, like an onion. The innermost layer holds business logic. Outer layers coordinate and expose that logic to the outside world.
+== Akka project structure
 
-image::docs-onion_architecture-v1.min.svg[Layered Structure of a Service, width=400]
+A typical Akka project might have a layout like the following:
 
-Each layer should live in its own package:
+[source,txt]
+----
+src/
+ ├── main/
+ │   ├── java/acme/planningagent/
+ │   │   ├── api/           # External MCP, HTTP, gRPC endpoints
+ │   │   ├── application/   # Akka components: Agents, Workflows, Entities, etc.
+ │   │   └── domain/        # Business logic
+ │   └── resources/
+ └── test/
+----
+
+* The `api` directory exposes functionality to the outside world. This includes HTTP, gRPC, or MCP interfaces that forward requests to the application layer.
+* The `application` directory contains the building blocks provided by Akka, implemented by you. It includes components such as `Agent`, `Entity`, `View`, `Workflow`, `Timer`, and `Consumer`.
+* The `domain` directory holds plain Java classes that describe business rules and domain models. These are not tied to Akka or the runtime. Many use `record` to reduce boilerplate. You can test this logic without starting Akka or the runtime. This keeps the code focused and easier to maintain.
+* The `resources` directory includes configuration files and other static content.
+* The `test` directory contains unit and integration tests. Its structure mirrors `main` to make it easier to relate tests to the code they verify.
+
+Keeping these areas distinct can help improve clarity and long-term maintainability. It also encourages testing and runtime separation.
 
-* `domain`: for the domain model
-* `application`: for Akka components
-* `api`: for external-facing endpoints
+== Conceptual layers
 
-Try not to expose domain types directly to the outside world. The API layer should not call the domain layer. Inner layers should not depend on or be aware of outer layers.
+The structure above also reflects a conceptual separation of responsibilities. These responsibilities can be thought of as layers. Business logic is central, with supporting code around it to enable runtime behavior and external interaction.
 
-== Layer overview
+To maintain modularity:
```

Comment:
We might also want to provide a link to the coding guidelines here, e.g. "For more details on best practices for more than just modules, check out the code guidelines"